### PR TITLE
Initial update to v2

### PR DESCRIPTION
**Pull request by CptSpaceToaster**
## V2 is here!

Please test this in a separate dev environment. I ran the changes in a quick test environment and updated the tests, but didn't _actually_ test things rigorously. Thanks :)

Things seemed to be functional

#### TODO:
- [ ] It seems like we'll want to think about making the advanced query system a little easier. V2 uses one query parameter named `q` instead of a bunch of query parameters. (I have no immediate plans here... but something to think about)
    - The other SDK's haven't been doing this however... so we're fine to leave it as-is IMO.
- [ ] There is also no mechanism to instantiate a client to hold an `X-Api-Key`... however that doesn't seem like a HUGE thing for a client-facing library to have. Maybe we can just ignore that for a bit?
- [ ] Update the examples in the README

### CODE REVIEW

File: coverage/client.test.ts [hunk 1]
```diff
@@ -13,15 +13,15 @@ describe('Client', () => {
     Client.get('cards', params)
       .then(response => {
         expect(response).to.be.a('array');
-        expect(response[0].name).to.equal('Gardevoir');
+        expect(response[0].name).to.equal('Ampharos');
```
Comment (CptSpaceToaster) [inline coverage/client.test.ts:16]: This seems like a bad test. Perhaps it could check to see if it got a _nonzero_ amount of cards back?

Comment (glinesbdev) [reply coverage/client.test.ts:16]: Yeah, something like this?

```typescript
expect(response.length).to.have.lengthOf.above(0);
```

**Comment by glinesbdev** [comment]
I ran the tests and they all passed and the code looks good. Also, I did start on a V2 port of this project but I couldn't get around to finishing it as life has gotten in the way. Maybe someone else can look into getting that started?

**Comment by CptSpaceToaster** [comment]
> I ran the tests and they all passed and the code looks good.
Awesome
> Maybe someone else can look into getting that started?
What were you thinking?

**Pull request merged**